<?xml version="1.0"?>

<launch>

  <arg name="with_human" default="false"/>

  <arg name="mongo_db" default="test"/>
  <param name="mongo_db" value="$(arg mongo_db)"/>
  <arg name="mongo_properties_collection" default="test_properties"/>
  <param name="mongo_properties_collection" value="$(arg mongo_properties_collection)"/>
  <arg name="mongo_results_collection" default="test_results"/>
  <param name="mongo_results_collection" value="$(arg test_results)"/>

  <param name="autosync" value="false" />
  <param name="go_home_after_execution" value="false" />

  <node name="robot_task_interface" pkg="task_planner_interface" type="robot_node"  output="screen" >
    <param name="topic_request_from_planner" value="/sharework/test/stiima/motion" />
    <param name="topic_feedback_to_planner" value="/sharework/test/stiima/motion/feedback" />
    <param name="group_name" value="manipulator" />
    <param name="retry_position" value="home" />
  </node>

  <node if="$(arg with_human)" name="human_task_interface" pkg="task_planner_interface" type="robot_node" output="screen" >
    <param name="topic_request_from_planner" value="/sharework/test/stiima/human" />
    <param name="topic_feedback_to_planner" value="/sharework/test/stiima/human/feedback" />
    <param name="group_name" value="human_right_arm" />
    <param name="retry_position" value="home" />
  </node>

  <node name="syncronization_node" pkg="task_planner_interface" type="syncronization_node"  output="screen" >
    <param name="number_of_agents" value="2"/>
  </node>


</launch>
